contracts/l2/L2Registry.sol

```diff
@@ -186,7 +260,7 @@ contract L2Registry is IERC1155 {
             return IController(address(0));
         }
         assembly {
-            addr := shr(96, mload(add(data, 32)))
+            addr := mload(add(data, 20))
```

Comment:
The upper bits of the address need to be masked out. I think due to that, the shr approach is probably just as efficient.